### ISSUE TITLE
config: Design of LoadDefaultConfig options is very generic, allows invalid loading options

### ISSUE THREAD

**Issue by jasdel** [aws/aws-sdk-go-v2#926]
The v2 SDK's design for `LoadDefaultConfig` options are very generic, and will accept any value since `Config` is an alias for `interface{}`. This means there are no direct links between the configuration loading utility and the options a user can provide. Other than looking for `With...` types/functions in the `config` module.  In addition, since `Config` is an empty interface, users could provide any value here regardless if it is valid and usable or not.

To address these issues, I propose the `config.LoadDefaultConfig` function signature be changed to take function options for a `LoadOptions` struct instead of abstract `Config` empty interface type. Instead of LoadDefaultConfig taking an abstract set of configuration provider getters, LoadDefaultConfig would take functional options of a discrete set of options that are valid for loading the configuration. Helpers like `WithRegion`, `WithSharedConfigProfile`, etc would be functional options that set fields on the `LoadOptions` struct instead of value getters.

Usage:
```go
cfg, err := config.LoadDefaultConfig(config.WithRegion("us-west-2"),
     func(o *config.LoadOptions) {
          o.SharedConfigProfile = "foo"
     })
```

config Package type definitions:
```go
type LoadOptions struct {
     aws.Config
     
     SharedConfigProfile string
     CustomCABundle io.Reader // or string?
}

func LoadDefaultConfig(optFns ...func(*LoadOptions)) (aws.Config, error) {}
```

The embedded `aws.Config` allows configuration of default configuration options such as Region, and EndpointResolver. the value of the embedded `aws.Config` would be initialized with the config's current `resolveDefaultConfig` function before the user provided option functions are invoked. This design also removes the needed for resolvers like `resolveEndpointResolvers`, `resolveRegion`, and `resolveHTTPClient`.

The `LoadOptions` type would implement the provider getters that are used by the config loading resolvers.

**Comment by jasdel** [comment]
`resolveRegion` resolver would still need to exist, since region can come from additional sources, such as environment variables and shared config. In addition to ec2 region if the SDK ever supports that.

**Comment by skotambkar** [comment]
Fixed in #951
